plugin-management-cli/src/main/java/io/jenkins/tools/pluginmanager/cli/CliOptions.java

```diff
@@ -57,4 +85,17 @@ public boolean isShowWarnings() {
     public boolean isShowAllWarnings() {
         return showAllWarnings;
     }
+
+    public String getJenkinsUc() {
+        return jenkinsUc == null ? "" : jenkinsUc.toString();
```

Comment:
I don't think there's any reason we should allow these to be un set

The other config value detection should all move here, env variable and default.

Reply:
See my response below. I was trying to keep the CLIOptions class only be for getting the CLIOptions. I ended up adding this line because I made this option be a URL so that checking for a malformed URL exception could be done in the CLI, but this is stored as a string up until the point of downloading the plugin so that additional info can easily be appended to it.